rfcs/proposed/task_group_wait_single_task/README.md

```diff
@@ -0,0 +1,496 @@
+
+// Submit additional work
+q.wait();
+```
+
+Consider implementing a similar model using ``tbb::task_group``. The queue can be represented as a ``task_group`` object, where tasks correspond to command groups in the queue, and events are modeled using ``tbb::task_completion_handle``s.
+
+```cpp
+using event = tbb::task_completion_handle;
+
+class queue {
+    template <typename Body>
+    event single_task(Body body) {
+        tbb::task_handle task = m_task_group.defer(body);
+        tbb::task_completion_handle comp_handle = task;
+        m_task_group.run(std::move(task));
+        return {comp_handle};
+    }
+
+    void wait() { m_task_group.wait(); }
+private:
+    tbb::task_group m_task_group;
+};
+```
+
+The ``sycl::handler::depends_on`` functionality can be implemented using the ``tbb::task_group::set_task_order`` API:
+
+```cpp
+class handler {
+    void depends_on(const std::vector<event>& events) {
+        for (event& e : events) {
+            tbb::task_group::set_task_order(e, m_root);
+        }
+    }
+
+    tbb::task_handle m_root;
+};
+```
+
+Since the example above also uses ``sycl::event::wait`` for wait for the completion of a single task (``task3``), implementing this behavior with
+a ``task_group`` requires support for waiting on a single ``task_completion_handle`` object.
+
+## Proposal
+
+The proposal aims to extend ``tbb::task_group`` and ``tbb::task_arena`` APIs by introducing new waiting functions that accept a
+``task_completion_handle`` argument.
+These functions exit when the task represented by the handle argument is completed or when the group execution is cancelled.
+The completion of other tasks within the ``task_group`` is not guaranteed.
+
+To simplify the submission and waiting process for a single task, it is also proposed to introduce submit-and-wait functions that accept a
+``task_handle`` argument.
+
+If task completion is transferred to another task using ``tbb::task_group::transfer_completion_to(other_task)``, the waiting functions will wait for
+the completion of ``other_task``.
+
+The current waiting functions return a ``task_group_status``, which can have three possible values:
+* ``tbb::task_group_status::not_complete`` - The group execution is not cancelled, and not all tasks in the group have completed.
+* ``tbb::task_group_status::complete`` - The group execution is not cancelled, and all tasks in the group have completed.
+* ``tbb::task_group_status::canceled`` - The group execution is cancelled, and the completion status of tasks is unknown.
+
+Since the new waiting functions track the progress of a single task, returning a ``task_group_status`` may be misleading.
+If the group execution is cancelled, the tracked task may still execute, and returning ``canceled`` does not accurately reflect the task's 
+completion status.
+If execution is not cancelled, the function would need to track whether other tasks remain in the group and return ``not_complete` if any are still pending.
```

Comment:
Will transfer of a task_completion lead to any confusion around cancellation, if for example, the original task has executed but the task that was transfered to is canceled?  The user will see canceled as a status even though the initial task did complete. I think it's fine and will just need to be well documented in transfer.